character-controller.js

```diff
@@ -1195,6 +1210,12 @@ class LocalPlayer extends UninterpolatedPlayer {
       this.characterHups.update(timestamp);
     }
   }
+  handleAnimationEnd(e) {

```

Comment:
This eventing + multi-frame solution seems nasty. Is there no way to do it cleanly in one function?

Reply:
Simplified logics: https://github.com/webaverse/app/pull/3010/commits/f596444cb71204c4cac9ad23f5bd63399b46d7eb , https://github.com/webaverse/app/pull/3010/commits/d98645faa5072b3ac63260c59f14d2d9c0b73289 .

And can even greatly simplify the logics by calling `game.handleAnimationEnd()` directly from avatar, instead of through avatar's `animationEnd` event. But not sure if this way is proper. https://github.com/webaverse/app/pull/3010/commits/f7518187c5c4f4e73b259dfcf6c55ac6970d1baa